src/components/SearchBarView.tsx

```diff
@@ -0,0 +1,135 @@
+import * as React from 'react';
+import { SafeAreaView, View, StyleSheet} from "react-native"
+import { Icon, Input } from 'react-native-elements';
+import { gql, useLazyQuery} from '@apollo/client'
+import * as GetTypes from '../typings/src/ts/types'
+
+// GraphQL fragment
+const COMPANY_TILE_DATA = gql` {
+    fragment CompanyTile on Business{
+        __typename
+        id
+        name
+        profilePicture
+        sustainabilityScore
+        customerScore
+    }
+}
+`
+;
+
+
+const SearchBar = () => {
+    // Search field
+    const [value, onChangeText] = React.useState('');
+    
+    // Query
+    const [executeSearch, {loading, error, data}] = useLazyQuery(
+        COMPANY_TILE_DATA
+      );
+
+
+    // MAYBE USE THIS: https://github.com/slorber/awesome-debounce-promise
+
+    // Debounce query
+    React.useEffect(() => {
+        const delayDebounceFn = setTimeout(() => {
+          executeSearch({
+              variables: {String: value}
+          })
+          console.log(data)
```

Comment:
When you executeSearch, you need to either use [async/await](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Async_await) or use [then](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises) to make sure that the promise has resolved without an error before moving on. If there is an error, it should probably be displayed to the user.